### PR TITLE
Toggle version diff in recent changes table

### PR DESCRIPTION
**Pull request by VirginiaDooley**
- [x] Toggle a row to view version diff
- [x] Improve the diff display (parses html to extract changed fields)
- [x] Store the fields edited on the logged action itself. 
- [x] Filtering by edits to a given field. We could then show this on the table for quick scanning
- [x] Fix rendering for new ballots and parties
- [x] Test matching for all version fields

How to test: 

1. Make a few candidate edits to populate db with version data; be sure to make an edit of `extra_fields` such as other name or biscuit (since this is not currently working with the current code in this PR)
2. http://localhost:8000/recent-changes
3. In advanced filters, select a filter from `VERSION FIELDS CONTAINS`

### CODE REVIEW

**Comment by VirginiaDooley** [comment]
This works well for Person fields, but adding a new ballot and/or party needs work. 
Screen Shot 2022-05-11 at 4 18 45 PM

**Comment by VirginiaDooley** [comment]
> This works well for Person fields, but adding a new ballot and/or party - which is recognised as a person update action - needs work. Screen Shot 2022-05-11 at 4 18 45 PM

I need to replace the entire ballot string from the version_fields, but currently, candidacies are not in version fields. Should they be added? If not, how else could I replace the ballot string from `diff_html`?

**Comment by VirginiaDooley** [comment]
The remaining work here is to get `version_fields` filtering working with `extra_fields/...`

File: ynr/apps/people/migrations/0039_auto_20220428_1635.py [hunk 1]
```diff
@@ -0,0 +1,21 @@
...
+
+    dependencies = [("people", "0038_remove_personimage_is_primary")]
+
+    operations = [
+        migrations.RemoveIndex(
+            model_name="person", name="name_vector_search_index"
+        ),
+        migrations.AddIndex(
+            model_name="person",
+            index=django.contrib.postgres.indexes.GinIndex(
```
Comment (michaeljcollinsuk) [inline ynr/apps/people/migrations/0039_auto_20220428_1635.py:17]: How did this change come about? Wondering if this is a missing migration from other work?

Comment (VirginiaDooley) [reply ynr/apps/people/migrations/0039_auto_20220428_1635.py:17]: @symroe Is this ok to commit with this PR, assuming it looks ok? I think it was missing from previous search-related work?

**Comment by VirginiaDooley** [comment]
Test failure at https://github.com/DemocracyClub/yournextrepresentative/blob/master/ynr/apps/people/tests/test_merge_view.py:L486: 
`Failed: Undefined template variable 'action.ballot.get_absolute_url' in '/Users/virginiadooley/Documents/development/yournextrepresentative/ynr/apps/candidates/templates/candidates/recent-changes.html'`. However, I don't get an error in the template/browser.

**Comment by VirginiaDooley** [comment]
[Add version diff to recent changes table](https://trello.com/c/O8dZmzFK/2943-add-version-diff-to-recent-changes-table)

**Review by symroe — APPROVED** [review]

**Pull request merged**